onert-micro/onert-micro/src/import/kernels/AddN.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "import/OMKernelConfigureBuilder.h"
+#include "core/OMUtils.h"
+#include "OMStatus.h"
+#include "execute/OMRuntimeKernel.h"
+
+using namespace onert_micro;
+using namespace onert_micro::core;
+
+namespace
+{
+
+constexpr uint32_t input1TensorIdx = 0;
+constexpr uint32_t outputTensorIdx = 0;
+
+} // namespace
+
+OMStatus onert_micro::import::configure_kernel_CircleAddN(const OMConfigureArgs &config_args)
+{
+  OMRuntimeContext &runtime_context = config_args.runtime_context;
+  uint16_t op_index = config_args.kernel_index;
+
+  onert_micro::execute::OMRuntimeKernel runtime_kernel;
+
+  OMStatus status = runtime_kernel.readKernel(op_index, runtime_context);
+  if (status != Ok)
+    return status;
+
+  const auto kInputsNum = runtime_kernel.inputs_num;
+
+  const circle::Tensor *output = runtime_kernel.outputs[outputTensorIdx];
+  assert(output != nullptr);
+  const circle::Tensor *input1 = runtime_kernel.inputs[input1TensorIdx];
+  assert(input1 != nullptr);
+  const OMRuntimeShape input1_shape(input1);
+
+  for (int idx = 1; idx < kInputsNum; ++idx)
+  {
+    const circle::Tensor *input = runtime_kernel.inputs[idx];
+    assert(input != nullptr);
+    status = utils::checkCondition(input1->type() == input->type());
+    if (status != Ok)
+      return status;
+  }
+
+  status = utils::checkCondition(input1->type() == output->type());
+  if (status != Ok)
+    return status;
+
+  for (int idx = 1; idx < kInputsNum; ++idx)
+  {
+    const circle::Tensor *input = runtime_kernel.inputs[idx];
+    assert(input != nullptr);
+    const OMRuntimeShape input_shape(input);
+    status = utils::checkCondition(input_shape.dimensionsCount() == input1_shape.dimensionsCount());
+    if (status != Ok)
+      return status;
+    for (int dim_idx = 0; dim_idx < input1_shape.dimensionsCount(); ++dim_idx)
+    {
+      status = utils::checkCondition(input_shape.dims(dim_idx) == input1_shape.dims(dim_idx));
+      if (status != Ok)
+        return status;
```

Comment:
better use just: `input_shape == input1_shape` due to OMRuntimeShape has overload operator ==.